### PR TITLE
br: add integration test for pitr

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: ref #47738

Problem Summary:
add integration test for pitr
### What is changed and how it works?
add integration test for pitr, currently only test for `delete range`
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/tests/br_pitr/run.sh [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+echo "wait checkpoint advance"
+sleep 10
+now_time=$(date "+%Y-%m-%d %H:%M:%S %z")
+echo "get the current time: $now_time"
+export GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/stream/only-checkpoint-ts-with-check=return(\"$now_time\")"
+set +e # we need to get the exit code of br
+i=0
+while true; do
+    # run br with failpoint to compare the current checkpoint ts with the current time
+    run_br --pd $PD_ADDR log status --task-name integration_test
```
Comment (YuJuncen) [inline br/tests/br_pitr/run.sh:59]: injecting a fail point is looks pretty hacky. is it possible to check the status with `jq`?

File: br/pkg/stream/rewrite_meta_rawkv.go [hunk 2]
```diff
@@ -708,7 +708,7 @@ func (sr *SchemasReplace) deleteRange(job *model.Job) error {
 	dbReplace, exist := sr.DbMap[job.SchemaID]
 	if !exist {
 		// skip this mddljob, the same below
-		log.Debug("try to drop a non-existent range, missing oldDBID", zap.Int64("oldDBID", job.SchemaID))
+		log.Warn("[rewrite delete range] try to drop a non-existent range, missing oldDBID", zap.Int64("oldDBID", job.SchemaID))
```
[the remark this answers is not in the document]
Comment (YuJuncen) [reply br/pkg/stream/rewrite_meta_rawkv.go:711]: (So perhaps we can add a field to the context and implicitly add it to each logs with `logutil.CL`)

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

**Review by YuJuncen — APPROVED** [review]

File: br/tests/br_pitr/run.sh [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+restart_services
+
+# PITR restore
+echo "run pitr"
+run_br --pd $PD_ADDR restore point -s "local://$TEST_DIR/$PREFIX/log" --full-backup-storage "local://$TEST_DIR/$PREFIX/full" > $res_file 2>&1
+
+# check something in downstream cluster
+echo "check br log"
+check_contains "restore log success summary"
+# check_not_contains "rewrite delete range"
```
Comment (Leavrth) [inline br/tests/br_pitr/run.sh:96]: Notice: After release-7.1, there are duplicate `delete-range`s when drop database whose table has partition. I would modified it in the cherry-pick to release-6.5 and fix it in the next PR.

File: br/pkg/stream/rewrite_meta_rawkv.go [hunk 4]
```diff
@@ -787,18 +789,19 @@ func (sr *SchemasReplace) deleteRange(job *model.Job) error {
...
+						zap.Int64("oldPartitionID", oldPid))
 					continue
 				}
-				physicalTableIDs[i] = newPid
+				newPhysicalTableIDs = append(newPhysicalTableIDs, newPid)
 			}
-			if len(physicalTableIDs) > 0 {
-				sr.insertDeleteRangeForTable(newJobID, physicalTableIDs)
+			if len(newPhysicalTableIDs) > 0 {
+				sr.insertDeleteRangeForTable(newJobID, newPhysicalTableIDs)
```
Comment (3pointer) [inline br/pkg/stream/rewrite_meta_rawkv.go:804]: maybe add some info logs inside this function to record which id should be deleted.

**Review by 3pointer — APPROVED** [review]

**Pull request merged**